File: clients/java3/src/org/openamq/client/state/AMQStateChangedEvent.java
package org.openamq.client.state;

/**
 * An event that is fired when the protocol state has changed.
 * 
 * @author Robert Greig (robert.j.greig@jpmorgan.com)
 */
public class AMQStateChangedEvent
{
    private final AMQState _oldState;

    private final AMQState _newState;

    public AMQStateChangedEvent(AMQState oldState, AMQState newState)
    {
        _oldState = oldState;
        _newState = newState;
    }

    public AMQState getOldState()
    {
        return _oldState;
    }

    public AMQState getNewState()
    {
        return _newState;
    }
}

File: dotnet/xmsclient/Client/State/AMQStateChangedEvent.cs
using System;
using jpmorgan.mina.common;
using log4net;

namespace OpenAMQ.XMS.Client.State
{
    public class AMQStateChangedEvent
    {
        private readonly AMQState _oldState;

        private readonly AMQState _newState;

        public AMQStateChangedEvent(AMQState oldState, AMQState newState)
        {
            _oldState = oldState;
            _newState = newState;
        }

        public AMQState OldState
        {
            get
            {
                return _oldState;
            }
        }

        public AMQState NewState
        {
            get
            {
                return _newState;
            }
        }

    }
}
